### PR TITLE
add video settings for HDR tone mapping

### PR DESCRIPTION
**Pull request by FernetMenta**
see title

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp [hunk 1]
```diff
@@ -1285,6 +1293,9 @@ void CLinuxRendererGL::RenderToFBO(int index, int field, bool weave /*= false*/)
   m_pYUVShader->SetHeight(planes[0].texheight);
   m_pYUVShader->SetNonLinStretch(1.0);
   m_pYUVShader->SetColParams(buf.m_srcColSpace, buf.m_srcBits, !buf.m_srcFullRange, buf.m_srcTextureBits);
+  m_pYUVShader->SetDisplayMetadata(buf.hasDisplayMetadata, buf.displayMetadata,
+                                   buf.hasLightMetadata, buf.lightMetadata);
+  m_pYUVShader->SetTomeMapParam(m_videoSettings.m_ToneMapParam);
```
Comment (wsnipex) [inline xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp]: typo: TomeMap

File: xbmc/video/VideoDatabase.cpp [hunk 2]
```diff
@@ -5286,11 +5297,17 @@ void CVideoDatabase::UpdateTables(int iVersion)
     m_pDS->exec("DROP TABLE settings");
     m_pDS->exec("ALTER TABLE settingsnew RENAME TO settings");
   }
+
+  if (iVersion < 110)
+  {
+    m_pDS->exec("ALTER TABLE settings ADD TonemapMethod integer");
+    m_pDS->exec("ALTER TABLE settings ADD TonemapParam float");
```
[the remark this answers is not in the document]
Comment (FernetMenta) [reply xbmc/video/VideoDatabase.cpp]: how exactly? my SQL skills are a bit rusty

[the remark this answers is not in the document]
Comment (wsnipex) [reply xbmc/video/VideoDatabase.cpp]: this should work:
"ALTER TABLE settings ADD TonemapMethod integer, TonemapParam float"

**Comment by afedchin** [comment]
@FernetMenta https://github.com/FernetMenta/xbmc/pull/476

File: xbmc/video/VideoDatabase.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (DaveTBlake) [reply xbmc/video/VideoDatabase.cpp]: Probably wise. IIRC there are differences between SQLite and MySQL syntax, these get allowed before the SQL is executed but our parsing may not be able to cope with the combined statement. But that is just my vague memory, so don't quote me!

File: xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShaderGL.cpp [hunk 3]
```diff
@@ -176,6 +176,8 @@ bool BaseYUV2RGBGLSLShader::OnEnabled()
     else if (m_hasDisplayMetadata && m_displayMetadata.has_luminance)
       param = log10(100) / log10(m_displayMetadata.max_luminance.num/m_displayMetadata.max_luminance.den);
```
Comment (fritsch) [inline xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShaderGL.cpp]: ``` m_displayMetadata.max_luminance.num/m_displayMetadata.max_luminance.den ```
Take care this is an integer division and results in zero if den < num. Btw. I am not sure the algorithm is correct. I have far better results when using lum_max / lum_min

**Comment by FernetMenta** [comment]
win64 build error unrelated

**Pull request merged**